File: RecyclerView_Swipe_Menu/app/src/main/java/kobe/recyclerview_swipe_menu/MyRecyclerViewAdapter.java
package kobe.recyclerview_swipe_menu;

import android.content.Context;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

/**
 * Created by kobe on 19/12/2017
 */

public class MyRecyclerViewAdapter extends RecyclerView.Adapter<MyRecyclerViewAdapter.BaseViewHolder> {

    private static final String TAG = "KKD";
    private Context mContext;
    private ArrayList<ItemData> mData;

    MyRecyclerViewAdapter(final Context context) {
        mContext = context;
        mData = new ArrayList<>();

        for(int i = 0; i < 50; i++) {
            mData.add(new ItemData("Text " + i));
        }
    }

    class BaseViewHolder extends RecyclerView.ViewHolder {
        TextView mTextView;

        BaseViewHolder(View itemView) {
            super(itemView);

            mTextView = itemView.findViewById(R.id.tv_content);
        }
    }

    void onItemDelete(final int position) {
        mData.remove(position);
        notifyItemRemoved(position);
    }

    void release() {
        mData.clear();
        mData = null;
        mContext = null;
    }

    @Override
    public BaseViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new BaseViewHolder(LayoutInflater.from(mContext).inflate(R.layout.layout_content, parent, false));
    }

    @Override
    public void onBindViewHolder(BaseViewHolder holder, int position) {
        holder.mTextView.setText(mData.get(position).getText());
    }

    @Override
    public int getItemCount() {
        return mData.size();
    }
}

File: RecyclerView_In_ScrollView/app/src/main/java/com/kobe/recyclerview_in_scrollview/MyRecyclerViewAdapter.kt
package com.kobe.recyclerview_in_scrollview

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import kotlinx.android.synthetic.main.item_layout.view.*

class MyRecyclerViewAdapter(private val context: Context): RecyclerView.Adapter<MyRecyclerViewAdapter.BaseViewHolder>() {

    private val items = ArrayList<Int>()

    init {
        for (i:Int in 0..100) {
            items.add(i)
        }
    }

    class BaseViewHolder(itemView: View?) : RecyclerView.ViewHolder(itemView) {
        val titleView: TextView? = itemView?.titleTextView
        val contentView: TextView? = itemView?.contentTextView
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder {
        return BaseViewHolder(LayoutInflater.from(context).inflate(R.layout.item_layout,parent, false))
    }

    override fun getItemCount(): Int {
        return items.size
    }

    override fun onBindViewHolder(holder: BaseViewHolder, position: Int) {
        val titleStr: String = "Title " + items[position]
        val contentStr: String = "Content " + items[position]

        holder.titleView?.text = titleStr
        holder.contentView?.text = contentStr
    }
}